lib/account/account-service.js

```diff
@@ -1,99 +1,140 @@
+        return getDirectoryAsync(registrationDirName, regId).then((regDirectory) => {
+            return getAsync(regDirectory).then((accountFile) => {
+                accountFile.accountId = accountInfo.accountId;
+                accountFile.password = accountInfo.password;
+                delete accountFile.regId;
 
-            return this.storage.putAsync(directory, JSON.stringify(this.accountFile), options).then(() => {
-                return {
-                    accountId: accountFile.accountId
-                };
+                return makeAccountFileAsync(accountFile, options).then(() => {
+                    return storage.delAsync(regDirectory).then(() => {
+                        return {
+                            accountId: accountInfo.accountId
+                        };
+                    });
+                });
             });
         });
     }
 
 
     /**
+     * Creats the account file in the account directory.
+     *
+     * @param {Object} accountInfo
+     * @param {Object} options
+     * @return {Promise.<boolean>}
+     */
+    function makeAccountFileAsync(accountInfo, options) {
+        return getDirectoryAsync(accountDirName, accountInfo.accountId).then((accountDir) => {
+            return storage.putAsync(accountDir, JSON.stringify(accountInfo), options);
+        });
+    }
+
+
+    /**
      * Gets the account file from the directory.
      *
      * TODO: Decrypt data
      *
-     * @param {string} accountId
+     * @param {string} directory
      * @return {Promise.<AccountService~accountFile>}
      */
-    getAsync(directory) {
-        return this.storage.getAsync(directory).then((accountFile) => {
+    function getAsync (directory) {
+        return storage.getAsync(directory).then((accountFile) => {
             return JSON.parse(accountFile.toString("binary"));
         });
     }
 
 
     /**
+     * Get the registration file when it's still in the registration directory.
+     * This is to eliminate one steps for the manager getting the directory
+     * and then requesting the file.
+     *
+     * @param {string} regId
+     * @return {Promise.<accoutService~accountFile>}
+     */
+    function getRegistrationFileAsync (regId) {
+        return getDirectoryAsync(registrationDirName, regId).then((directory) => {
+            return getAsync(directory).then((accountFile) => {
+                return accountFile;
+            });
+        });
+    }
+
+
+    /**
      * Gets the directory where the account will be/is stored.
      *
-     * @param {string} accountId
+     * @param {string} prefix
+     * @param {string} idToHash
      * @return {Promise.<string>}
      */
-    getDirectoryAsync(accountId) {
-        return this.secureHash.hashAsync(accountId, this.accountConfig.idHash || {}).then((hashedAccountId) => {
-            return this.directoryName + hashedAccountId;
+    function getDirectoryAsync (prefix, idToHash) {
```

Comment:
No space before the open parenthesis.  (not mentioning it again, though it is in other spots)